src/studio/src/designer/frontend/ux-editor/containers/DroppableDraggableComponent.tsx

```diff
@@ -1,316 +1,101 @@
-            if (hoverOverIndex === draggedContainer.index) {
-              return;
-            }
-          }
-
-          props.onMoveContainer(
-            draggedContainer.id,
-            hoverOverIndex,
-            props.containerId,
-            component.props.containerId,
-          );
-
-          draggedContainer.index = hoverOverIndex;
-          break;
-        }
-        default: {
-          break;
-        }
-      }
+    if (!monitor.isOver({ shallow: true })) {
+      return; // we are not over... do nothing
+    }
+    if (
+      !hoverIndexHelper(draggedItem, targetItem, ref, monitor.getClientOffset())
+    ) {
+      return; // we are not performing any actions
     }
+    events.moveItem(draggedItem, targetItem);
   },
-};
+});
 
-class DroppableDraggableComponent extends React.Component<
-  IDroppableDraggableComponentProps & {
-    connectDragPreview: ConnectDragPreview;
-    connectDragSource: ConnectDragSource;
-    connectDropTarget: ConnectDropTarget;
-    dropTargetMonitor: DropTargetMonitor;
-    isDragging: boolean;
-  },
-  any
-> {
-  public render() {
-    const {
-      id,
-      connectDropTarget,
-      connectDragPreview,
-      connectDragSource,
-      isDragging,
-    } = this.props;
-    const style = isDragging
-      ? {
-          opacity: 0,
-        }
-      : null;
-    return connectDropTarget(
-      connectDragPreview(
-        connectDragSource(
-          <div key={id} style={style}>
-            {this.props.children}
-          </div>,
-        ),
-      ),
-    );
-  }
+export interface IDroppableDraggableComponentProps {
+  canDrag: boolean;
+  children?: React.ReactNode;
+  containerId: string;
+  dndEvents: EditorDndEvents;
+  id: string;
+  index: number;
 }
 
-export default DropTarget(
-  ['ITEM', 'TOOLBAR_ITEM', 'CONTAINER'],
-  dropTargetSpec,
-  (connect: DropTargetConnector, monitor: DropTargetMonitor) => ({
-    connectDropTarget: connect.dropTarget(),
-    dropTargetMonitor: monitor,
-  }),
-)(
-  DragSource(
-    'ITEM',
-    dragSourceSpec,
-    (connect: DragSourceConnector, monitor: DragSourceMonitor) => ({
-      connectDragSource: connect.dragSource(),
-      connectDragPreview: connect.dragPreview(),
-      isDragging: monitor.isDragging(),
-    }),
-  )(DroppableDraggableComponent),
-);
+export const DroppableDraggableComponent: React.FC<IDroppableDraggableComponentProps> =
+  memo(function DroppableDraggableComponent({
+    canDrag,
+    children,
+    containerId,
+    dndEvents,
+    id,
+    index,
+  }: IDroppableDraggableComponentProps) {
+    const ref = useRef<HTMLDivElement>(null);
+
+    const item = { id, containerId, index, type: ItemType.ITEM };
+    // eslint-disable-next-line no-empty-pattern
+    const [{ isDragging }, drag] = useDrag(
+      dragSourceSpec(item, canDrag, dndEvents.onDropItem),
+    );
+
+    // eslint-disable-next-line no-empty-pattern
+    const [{}, drop] = useDrop(dropTargetSpec(item, dndEvents, ref));
```

Comment:
Refactor to not use eslint statement 😇

```suggestion
    const result = useDrop(dropTargetSpec(item, dndEvents, ref));
    const drag = result.drag;
```